core/src/main/java/org/jruby/RubyDir.java

```diff
@@ -326,6 +326,9 @@ public static IRubyObject chdir(ThreadContext context, IRubyObject recv, IRubyOb
         if (adjustedPath.startsWith("uri:")){
             realPath = adjustedPath;
         }
+        else if (adjustedPath.startsWith("file:")){
+            realPath = "uri:" + adjustedPath;
+        }
```

Comment:
I think we all to chdir to all paths following this pattern: https://github.com/jruby/jruby/blob/jruby-1_7/core/src/main/java/org/jruby/RubyFile.java#L670

it feels wrong to switch one pseudo-uri to another